hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMStateMachine.java

```diff
@@ -179,25 +193,79 @@ public long takeSnapshot() throws IOException {
     long startTime = Time.monotonicNow();
     TermIndex lastTermIndex = getLastAppliedTermIndex();
     long lastAppliedIndex = lastTermIndex.getIndex();
-    TransactionInfo lastAppliedTrxInfo =
-        TransactionInfo.fromTermIndex(lastTermIndex);
-    if (transactionBuffer.getLatestTrxInfo()
-        .compareTo(lastAppliedTrxInfo) < 0) {
+    if (isInitialized.get()) {
```

Comment:
Here there is a chance of missing DB update.

Lets take 1,2,4 metadata/confUpdate and 3 is applyTransaction (Still not added to TransactionDBBuffer)

And now when **takeSnapshot** lastAppliedTermIndex will be 4, (As notifyConfigurationChanged will be called) and we flush. Restart SCM, we think 4 is applied but it is not.

Reply:
This cannot happen. applyTransaction is executed in the log order and applTransaction completely sequential in SCM stateMachine.